lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc

```diff
@@ -396,12 +396,21 @@ function dosomething_campaign_preprocess_action_page(&$vars, &$wrapper) {
   // Pass the signup id to the front end.
   if (module_exists('dosomething_signup')) {
     if ($sid = dosomething_signup_exists($vars['nid'])) {
+      // Get competition data
+      $signup = signup_load($sid);
+      $competition = ['joined' => $signup->competition];
+      if ($competition['joined']) {
+        $signup_form = dosomething_signup_get_signup_data_form_info($signup->nid);
+        $competition['confirm_msg'] = $signup_form['confirm_msg'];
+      }
+
       drupal_add_js(
-        array('dosomethingSignup' =>
-          array(
+        ['dosomethingSignup' =>
+          [
             'sid' => $sid,
-          ),
-        ),
+            'competition' => $competition
```

Comment:
this isn't a competition object or id or anything right? just the confirmation message? Maybe structure it as such, so 

```php
`competition` => [
  'messages' => [
    'confirmation' => '$confirmation_message,
  ]
]
```

it's a bit more verbose, but if we need to expand on this object later it will afford us the opportunity to do so! 

Up to you if you want to make `messages` array, maybe it's just one property as `confirmation_message` or something and not a nested array inside the `competition` array.